src/highdicom/legacy/sop.py

```diff
@@ -417,66 +545,1911 @@ def __init__(
+            src_kw_or_tg = tag_for_keyword(src_kw_or_tg)
+        if dest_kw_or_tg is None:
+            dest_kw_or_tg = src_kw_or_tg
+        elif isinstance(dest_kw_or_tg, str):
+            dest_kw_or_tg = tag_for_keyword(dest_kw_or_tg)
+        if check_not_to_be_perframe:
+            if src_kw_or_tg in self._perframe_tags:
+                return
+        if src_kw_or_tg in src_ds:
+            elem = src_ds[src_kw_or_tg]
+            if check_not_to_be_empty:
+                if self._is_empty_or_empty_items(elem):
+                    return
+            new_elem = deepcopy(elem)
+            if dest_kw_or_tg == src_kw_or_tg:
+                dest_ds[dest_kw_or_tg] = new_elem
+            else:
+                new_elem1 = DataElement(dest_kw_or_tg,
+                                        dictionary_VR(dest_kw_or_tg),
+                                        new_elem.value)
+                dest_ds[dest_kw_or_tg] = new_elem1
+            # now mark the attrib as used/done to keep track of every one of it
+            self._mark_tag_as_used(src_kw_or_tg)
+
+    def _get_or_create_attribute(
+        self, src: Dataset, kw: Union[str, Tag], default: Any) -> DataElement:
+        """Creates a new DataElement with a value copied from the source
+        Dataset. If the attribute is absent in source Dataset, then its value
+        will be the default value.
+        Parameters
+        ----------
+        src: Dataset
+            Source Dataset to copy the value if available.
+        kw: Union[str, Tag]
+            The keyword of created DataElement.
+        default: Any
+            The default value created DataElement if the keyword was absent in
+            the source Dataset.
+        Returns
+        -------
+        A new DataElement created.
+        """
+        if kw is str:
+            tg = tag_for_keyword(kw)
+        else:
+            tg = kw
+        if kw in src:
+            a = deepcopy(src[kw])
+        else:
+            a = DataElement(tg, dictionary_VR(tg), default)
+        from pydicom.valuerep import DT, TM, DA
+        if a.VR == 'DA' and isinstance(a.value, str):
+            try:
+                d_tmp = DA(a.value)
+                a.value = DA(default) if d_tmp is None else d_tmp
+            except BaseException:
+                a.value = DA(default)
+        if a.VR == 'DT' and isinstance(a.value, str):
+            try:
+                dt_tmp = DT(a.value)
+                a.value = DT(default) if dt_tmp is None else dt_tmp
+            except BaseException:
+                a.value = DT(default)
+        if a.VR == 'TM' and isinstance(a.value, str):
+            try:
+                t_tmp = TM(a.value)
+                a.value = TM(default) if t_tmp is None else t_tmp
+            except BaseException:
+                a.value = TM(default)
+
+        self._mark_tag_as_used(tg)
+        return a
+
+    def _add_module(self, module_name: str, excepted_attributes: list = [],
```

Comment:
Please follow the style of the package and place each parameter on a separate line if the length of the function definition would exceed the character limit.